skbio/tree/_nj.py

```diff
@@ -175,10 +175,11 @@ def _compute_q(dm):
     """
     q = np.zeros(dm.shape)
     n = dm.shape[0]
-    for i in range(n):
-        for j in range(i):
-            q[i, j] = q[j, i] = \
-                ((n - 2) * dm[i, j]) - dm[i].sum() - dm[j].sum()
+    dmv = dm.to_data_frame().values
```

Comment:
In this specific piece of code, using `dm.data` is definitely faster, particularly for distance matrices larger than 500x500. I will incorporate this change into #1513. Thanks!